### PR TITLE
Migrate defaultFiltersTests and issue55Tests groovy->java

### PR DESCRIPTION
**Pull request by kostya05983**
I moved defaultFiltersTests and issue55Tests to java from groovy, you can see gh-4939
@rwinch , @jzheaux please see:)

### CODE REVIEW

File: config/src/test/java/org/springframework/security/config/annotation/web/configuration/BaseWebConfig.java [hunk 1]
```diff
@@ -15,26 +15,25 @@
...
- *
  * @author Rob Winch
+ * @author Konstantin Volivach
  */
 @EnableWebSecurity
 public abstract class BaseWebConfig extends WebSecurityConfigurerAdapter {
-	BaseWebConfig(boolean disableDefaults) {
-		super(disableDefaults)
+
+	public BaseWebConfig(boolean disableDefaults) {
```
Comment (jzheaux) [inline config/src/test/java/org/springframework/security/config/annotation/web/configuration/BaseWebConfig.java]: Ideally, we wouldn't add this class. Would you instead be able to add to the test files something like:

```java
@Configuration
static class UserDetailsServiceConfig {
	@Bean
	public UserDetailsService userDetailsService() {
		return new InMemoryUserDetailsManager(PasswordEncodedUser.user());
	}
}
```

And then use `SpringTestRule` in the tests to instead do:

```java
this.spring.register(MyTestConfig.class, UserDetailsServiceConfig.class).autowire();
```

File: config/src/test/java/org/springframework/security/config/annotation/web/configurers/Issue55Tests2.java [hunk 2]
```diff
@@ -1,50 +1,57 @@
...
-	def "WebSecurityConfigurerAdapter defaults to @Autowired"() {
-		setup:
-			TestingAuthenticationToken token = new TestingAuthenticationToken("test", "this")
-		when:
-			loadConfig(WebSecurityConfigurerAdapterDefaultsAuthManagerConfig)
-		then:
-			context.getBean(FilterChainProxy)
-			findFilter(FilterSecurityInterceptor).authenticationManager.authenticate(token) == CustomAuthenticationManager.RESULT
-	 }
+public class Issue55Tests2 extends BaseSpringSpec2 {
```
[the remark this answers is not in the document]
Comment (kostya05983) [reply config/src/test/java/org/springframework/security/config/annotation/web/configurers/Issue55Tests2.java]: It's just my mistake, I renamed to Issue55Tests

File: config/src/integration-test/groovy/org/springframework/security/config/annotation/authentication/ldap/LdapAuthenticationProviderBuilderSecurityBuilderTests.groovy [hunk 3]
```diff
@@ -15,27 +15,18 @@
  */
 package org.springframework.security.config.annotation.authentication.ldap
 
-import org.springframework.beans.factory.config.AutowireCapableBeanFactory
```
[the remark this answers is not in the document]
Comment (kostya05983) [reply config/src/integration-test/groovy/org/springframework/security/config/annotation/authentication/ldap/LdapAuthenticationProviderBuilderSecurityBuilderTests.groovy]: Leave it untouched and others such classes

File: config/src/test/java/org/springframework/security/config/annotation/BaseSpringSpec2.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+import javax.servlet.http.HttpSession;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Rob Winch
+ * @author Konstantin Volivach
+ */
+public class BaseSpringSpec2 {
```
[the remark this answers is not in the document]
Comment (kostya05983) [reply config/src/test/java/org/springframework/security/config/annotation/BaseSpringSpec2.java]: I use springTestRule and delete this class

File: config/src/test/java/org/springframework/security/config/annotation/web/configuration/BaseWebConfig.java (hunk 1 above)
Comment (kostya05983) [reply config/src/test/java/org/springframework/security/config/annotation/web/configuration/BaseWebConfig.java]: Ok, I made as you say

**Comment by kostya05983** [comment]
@jzheaux I ended with changes. You can see. I made force push, that's why request change is remained.

**Comment by jzheaux** [comment]
Thanks, @kostya05983 for the PR! I've merged this via 950e6422a142b35f3ed21bf3dead36f41ecab805 along with a slight polish 1176d0cfdba189bde72c305d51ade15004e147d4

**Pull request closed without merging**